spec/plugin/go_spec.rb

```diff
@@ -106,4 +106,134 @@ def setup_go_filetype
       StructType{one: 1, two: "asdf", three: []int{1, 2, 3}}
     EOF
   end
+
+  describe "funcs" do
+    def assert_split_join(initial, split_expected, join_expected)
+      set_file_contents initial
+      setup_go_filetype
+      split
+      # In case there is no Go installed, deindent everything:
+      vim.normal '9<<9<<9<<9<<'
+      vim.write
+      assert_file_contents split_expected
+      join
+      assert_file_contents join_expected
+    end
+
+    it "handles function definitions" do
+      initial = <<-EOF
+        func Func(a, b int, c time.Time, d func(int) error, e func(int, int) (int, error), f ...time.Time) {
+        }
+      EOF
+      split = <<-EOF
+        func Func(
+        a, b int,
```

Comment:
This is what I want, not:

```
func Func(
a int,
b int,
...
```

But I don't know what others want. Perhaps we can provide an option for this.